engine/runtime/src/main/java/org/enso/interpreter/runtime/data/vector/ArrayProxy.java

```diff
@@ -30,25 +30,14 @@ final class ArrayProxy implements EnsoObject {
   private final long length;
   private final Object at;
 
-  private ArrayProxy(long length, Object at) throws IllegalArgumentException {
-    if (CompilerDirectives.inInterpreter()) {
```

Comment:
This caused the failure in https://github.com/enso-org/enso/actions/runs/7931422662/job/21655632171?pr=9065#step:8:6272 because the check was only done if the code was run in the interpreter. Once PEd, no more type errors were thrown. These are some nasty bugs - once you attach debugger and put a breakpoint into this method, it will not be PEd and thus, the error will not be manifested.